r10k.gemspec

```diff
@@ -18,23 +18,21 @@ Gem::Specification.new do |s|
     dynamic environments.
   DESCRIPTION
 
-  s.required_ruby_version = '>= 2.3.0'
+  s.required_ruby_version = '>= 2.5.0'
 
   s.license  = 'Apache-2.0'
 
   s.add_dependency 'colored2',   '3.1.2'
-  s.add_dependency 'cri', '2.15.10'
+  s.add_dependency 'cri', '>= 2.15.10'
 
   s.add_dependency 'log4r',     '1.1.10'
   s.add_dependency 'multi_json', '~> 1.10'
 
-  s.add_dependency 'puppet_forge', '~> 2.3.0'
+  s.add_dependency 'puppet_forge', '>= 2.3.0', '< 4'
 
   s.add_dependency 'gettext-setup', '~>0.24'
-  # These two pins narrow what is allowed by gettext-setup,
-  # to preserve compatability with Ruby 2.4
-  s.add_dependency 'fast_gettext', '~> 1.1.0'
-  s.add_dependency 'gettext', ['>= 3.0.2', '< 3.3.0']
+  s.add_dependency 'fast_gettext', '>= 1.1.0', '< 3'
+  s.add_dependency 'gettext', '>= 3.0.2', '< 4.0.0'
```

Comment:
Do we need these pins at all anymore?

Reply:
good question. for the upper pinning this is a good practice in many modules. Recently psych and faraday hat major releases with a lot of breaking changes. I think upper limits are fine, but I don't have a string opinion here. for the lower limits I am not sure whats the oldest working version / if we should only have an upper limit.

Reply:
I just meant that it looks like these are transitive dependencies via `gettext-setup`, rather than direct dependencies of r10k, I don't think we should pin them explicitly unless we need to. I would hope that `gettext-setup` does its own pinning to prevent breaking changes from its dependencies and that we could just pin `gettext-setup` if we need to.

Reply:
okay, I think this is good to go now.

Reply:
thanks so much!